search/searcher/geoshape_contains_test.go

```diff
@@ -0,0 +1,938 @@
+		Desc             string
+		QueryType        string
+	}{
+		{
+			QueryShape:       [][][][]float64{leftRect},
+			DocShapeVertices: leftRect,
+			DocShapeName:     "polygon1",
+			Expected:         []string{"polygon1"},
+			Desc:             "coincident polygons",
+			QueryType:        "contains",
+		},
+		{
+			QueryShape:       [][][][]float64{{{{2, 2}, {-2, 2}, {-2, -2}, {2, -2}}}},
+			DocShapeVertices: leftRect,
+			DocShapeName:     "polygon1",
+			Expected:         nil,
+			Desc:             "polygon larger than polygons in query shape",
+			QueryType:        "contains",
+		},
+	}
+
+	i := setupIndex(t)
+
+	for _, test := range tests {
+		indexReader, closeFn, err := testCaseSetup(t, test.DocShapeName, "polygon",
+			[][][][]float64{test.DocShapeVertices}, i)
+		if err != nil {
+			t.Errorf(err.Error())
+		}
+
+		t.Run(test.Desc, func(t *testing.T) {
+			got, err := runGeoShapeMultiPolygonQueryWithRelation(test.QueryType,
+				indexReader, test.QueryShape, "geometry")
+			if err != nil {
+				t.Errorf(err.Error())
+			}
+			if !reflect.DeepEqual(got, test.Expected) {
+				t.Errorf("expected %v, got %v for polygon: %+v",
+					test.Expected, got, test.QueryShape)
+			}
+		})
+
+		err = closeFn()
+		if err != nil {
+			t.Errorf(err.Error())
+		}
+	}
+}
+
+func TestMultiLinestringMultiPolygonContains(t *testing.T) {
+	tests := []struct {
+		QueryShape       [][][]float64
+		DocShapeVertices [][][][]float64
+		DocShapeName     string
+		Expected         []string
+		Desc             string
+		QueryType        string
+	}{
+		{
+			QueryShape:       [][][]float64{{{0.2, 1}, {0.8, 1}}, {{1, 0.2}, {1, 0.8}}},
+			DocShapeVertices: [][][][]float64{rightRect},
+			DocShapeName:     "multipolygon1",
+			Expected:         nil, // contains doesn't include edges or vertices
+			Desc:             "linestrings on edge of polygon",
+			QueryType:        "contains",
+		},
+		{
+			QueryShape:       [][][]float64{{{0.2, 0.2}, {0.8, 0.8}}, {{0.8, 0.2}, {0.2, 0.8}}},
+			DocShapeVertices: [][][][]float64{{{{0, 0}, {1, 0}, {1, 1}, {0, 1}, {0, 0}}}},
+			DocShapeName:     "multipolygon1",
+			Expected:         []string{"multipolygon1"},
+			Desc:             "linestrings within polygon",
+			QueryType:        "contains",
+		},
+	}
+
+	i := setupIndex(t)
+
+	for _, test := range tests {
+		indexReader, closeFn, err := testCaseSetup(t, test.DocShapeName, "multipolygon", test.DocShapeVertices, i)
+		if err != nil {
+			t.Errorf(err.Error())
+		}
+
+		t.Run(test.Desc, func(t *testing.T) {
+			got, err := runGeoShapeMultiLinestringQueryWithRelation(test.QueryType,
+				indexReader, test.QueryShape, "geometry")
+			if err != nil {
+				t.Errorf(err.Error())
+			}
+			if !reflect.DeepEqual(got, test.Expected) {
+				t.Errorf("expected %v, got %v for multilinestring: %+v",
+					test.Expected, got, test.QueryShape)
+			}
+		})
+		err = closeFn()
+		if err != nil {
+			t.Errorf(err.Error())
+		}
+	}
+}
+
+func TestGeometryCollectionPolygonContains(t *testing.T) {
```

Comment:
There could be a TestGeometryCollectionMultiPolygonContains test where as the various shapes inside the GC would be covered by multiple polygons in the multipolygon?